vapi/simulator/simulator.go

```diff
@@ -1339,6 +1339,16 @@ func (s *handler) libraryDeploy(lib *library.Library, item *item, deploy vcenter
 		}
 	}
 
+	if ds.Value == "" {
+		// Datastore is optional in the deploy spec, but not in OvfManager.CreateImportSpec
+		refs, err = v.Find(ctx, []string{"Datastore"}, nil)
+		if err != nil {
+			return nil, err
+		}
+		// TODO: consider StorageProfileID
+		ds = refs[0]
```

Comment:
TODO: Consider using  ds.AttachedHosts/AttachedClusterHosts to verify that selected datastore is attached to the target host/cluster.

Reply:
Agreed, we can improve the default datastore selection in this respect.  I'll merge for now as this fixes the panic in the common case of using `model.VPX`, where the default `LocalDS_0` is attached to all hosts.